sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -959,7 +959,16 @@ ProgramManager::getDeviceImage(OSModuleHandle M, KernelSetId KSId,
     debugPrintBinaryImages();
   }
   std::lock_guard<std::mutex> Guard(Sync::getGlobalLock());
-  std::vector<RTDeviceBinaryImageUPtr> &Imgs = *m_DeviceImages[KSId];
+  std::vector<RTDeviceBinaryImageUPtr> *ImgsPtr = nullptr;
+  {
+    auto It = m_DeviceImages.find(KSId);
+    if (It == m_DeviceImages.end())
+      throw runtime_error("No device image found for KernelSetId " +
+                              std::to_string(KSId) + " was found",
+                          PI_ERROR_INVALID_KERNEL_NAME);
+    ImgsPtr = It->second.get();
+  }
+  std::vector<RTDeviceBinaryImageUPtr> &Imgs = *ImgsPtr;
```

Comment:
I am Ok with the change. The only minor comment - it can be simplified/shortened by 3 lines:
```suggestion
  auto It = m_DeviceImages.find(KSId);
  if (It == m_DeviceImages.end())
    throw runtime_error("No device image found for KernelSetId " +
                            std::to_string(KSId) + " was found",
                        PI_ERROR_INVALID_KERNEL_NAME);
  std::vector<RTDeviceBinaryImageUPtr> *ImgsPtr = It->second.get();
  std::vector<RTDeviceBinaryImageUPtr> &Imgs = *ImgsPtr;
```
The last 2 lines also can be simplified more to:
  std::vector<RTDeviceBinaryImageUPtr> &Imgs = *It->second;
For additional verbosity it could be: 
  std::vector<RTDeviceBinaryImageUPtr> &Imgs = *(It->second);